npm/pkg/dataplane/dataplane_windows.go

```diff
@@ -50,13 +50,31 @@ func (dp *DataPlane) initializeDataPlane() error {
 		},
 		Flags: hcn.HostComputeQueryFlagsNone,
 	}
+	// Initialize Endpoint query used to filter healthy endpoints (vNIC) of Windows pods on L1VH Node
+	dp.endpointQueryL1VH.query = hcn.HostComputeQuery{
+		SchemaVersion: hcn.SchemaVersion{
+			Major: hcnSchemaMajorVersion,
+			Minor: hcnSchemaMinorVersion,
+		},
+		Flags: hcn.HostComputeQueryFlagsNone,
+	}
+
 	// Filter out any endpoints that are not in "AttachedShared" State. All running Windows pods with networking must be in this state.
 	filterMap := map[string]uint16{"State": hcnEndpointStateAttachedSharing}
+	filterMapL1VH := map[string]uint16{"State": hcnEndpointStateAttached}
+
 	filter, err := json.Marshal(filterMap)
 	if err != nil {
-		return npmerrors.SimpleErrorWrapper("failed to marshal endpoint filter map", err)
+		return errors.Wrap(err, "failed to marshal endpoint filter map")
+	}
+
+	filterL1VH, errL1VH := json.Marshal(filterMapL1VH)
+	if errL1VH != nil {
+		return errors.Wrap(errL1VH, "failed to marshal endpoint filter map")
```

Comment:
nit: Can we change the error message? Might make debugging easier one day

Reply:
yes added more detailed error messaged. Please let me know if that looks better